hkmc2/shared/src/test/mlscript/backlog/Lifter.mls

```diff
@@ -0,0 +1,149 @@
+:js
+:lift
+:todo
+
+// We use the optional `symbol` parameter of `Select` to detect references to the
+// BlockMemberSymbol. But when this symbol is not present, there is no way to properly
+// detect it. What do to?
```

Comment:
This could be lifted as, essentially:
```fsharp
class B$(y)(a) with
  fun getB() = a.x + y
class A(x) with
  fun B(y) = B$(y)(this)
  set B.class = B // so that `new a.B(n)` still works
  fun getA() = this.B(2).getB()
A(1).getA()
```
and we'd require that the class `B` be final, so it can't be used in `extends` clauses (that check can be done by the type checker later).

It can be done in a future PR. But please update the comment and document this in https://github.com/hkust-taco/mlscript/issues/285